connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/CheckpointSuite.scala

```diff
@@ -50,12 +51,20 @@ class CheckpointSuite extends ConnectFunSuite with RemoteSparkSession with SQLHe
     checkFragments(captureStdOut(block), fragmentsToCheck)
   }
 
-  test("checkpoint") {
+  test("localCheckpoint") {
     val df = spark.range(100).localCheckpoint()
     testCapturedStdOut(df.explain(), "ExistingRDD")
   }
 
-  test("checkpoint gc") {
+  test("localCheckpoint with StorageLevel") {
+    // We don't have a way to reach into the server and assert the storage level server side, but
+    // this test should cover for unexpected errors in the API.
```

Comment:
@hvanhovell with the SQL API refactoring, would it be now possible to have tests that use a connect client to self-connect, and have server side objects (SparkContext) etc. available inside the test to verify? The existing `SparkConnectServerTest` can only test internal SparkConnectClient with the server, due to past namespace conflicts between server and client SparkSession APIs.

Reply:
@juliuszsompolski it will take a few more PRs, but yeah that is the objective.